unic/utils/src/char_property.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+//! Taxonomy and Contracts for Character Property types.
+//!
+//! See also the list of types of character properties defined in the UCD:
+//! <http://unicode.org/reports/tr44/#About_Property_Table>, in [Unicode® Standard Annex #44 —
+//! Unicode Character Database](http://unicode.org/reports/tr44/#About_Property_Table)
+
+
+use std::fmt::{Debug, Display};
+use std::hash::Hash;
+
+
+/// A Character Property defined on all characters.
+///
+/// Examples: *Age*, *Name*, *General_Category*, *Bidi_Class*
+pub trait CharProperty
```

Comment:
I guess the only real example is the (ongoing?) discussion of an `impl<T: From> TryFrom for T`.

I'd be fine without this, but this would also simplify the marker for `Enumerated/NumericCharProperty` to just be `OptionCharProperty` since `CharProperty` implies `CharProperty`.

I don't really have an argument for it, but it just seems like "infallible transformation from char" is a subset of "fallible transformation from char". (Thus, the same use case as `Result<_, !>`?)